lib/openstudio-standards/schedules/parametric.rb

```diff
@@ -486,57 +486,80 @@ def self.gather_inputs_parametric_schedules(sch, load_inst, parametric_inputs, h
       # cleanup existing profiles
       OpenstudioStandards::Schedules.schedule_ruleset_cleanup_profiles(sch)
 
-      # gather profiles
-      daily_flhs = [] # will be used to tag, min,medium,max operation for non typical operations
-      schedule_days = {} # key is day_schedule value is rule index
-      sch.scheduleRules.each do |rule|
-        schedule_days[rule.daySchedule] = rule.ruleIndex
-        daily_flhs << OpenstudioStandards::Schedules.schedule_day_get_equivalent_full_load_hours(rule.daySchedule)
+      # get initial hash of schedule days => rule indices
+      schedule_days = schedule_ruleset_get_schedule_day_rule_indices(sch)
+      # get all day schedule equivalent full load hours to tag
+      daily_flhs = schedule_days.keys.map { |day_sch| schedule_day_get_equivalent_full_load_hours(day_sch) }
+      # collect initial rule index => array of days used hash
+      sch_ruleset_days_used = schedule_ruleset_get_annual_days_used(sch)
+
+      # match up schedule rule days with hours of operation days
+      sch_day_map = {}
+      sch_ruleset_days_used.each do |sch_index, sch_days|
+        day_map = {}
+        sch_days.each do |day|
+          # find the hour of operation rule that contains the day number
+          hoo_key = hours_of_operation.find { |_, val| val[:days_used].include?(day) }&.first
```

Comment:
The &.method was introduced in Ruby 2.3. Is there a way you can generalize it to allow earlier Ruby versions? We still support older ruby versions in openstudio-standards for all but a few sizing methods.